tests/e2e/config_secret_utils.go

```diff
@@ -0,0 +1,563 @@
+		clusterResult, err := sshExec(sshClientConfig, masterIp, cluster)
+		if err != nil && clusterResult.Code != 0 {
+			fssh.LogResult(clusterResult)
+			return nil, fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+				cluster, masterIp, err)
+		}
+		if clusterResult.Stdout != "" {
+			clusListTemp := strings.Split(clusterResult.Stdout, "\n")
+			clusList = append(clusList, clusListTemp...)
+		}
+		for i := 0; i < len(clusList); i++ {
+			if clusList[i] != "" {
+				clusterList = append(clusterList, clusList[i])
+			}
+		}
+		clusList = nil
+	}
+	return clusterList, nil
+
+}
+
+//getEsxiHostDetails method is used to fetch esxi hosts details
+func getEsxiHostDetails(vcAddress string, vCenterPort string, sshClientConfig *ssh.ClientConfig,
+	masterIp string, cluster []string) ([]string, error) {
+	var hostsList []string
+	var hostList []string
+	// Get all esxi hosts details
+	ginkgo.By("Executing command for fetching esxi host details")
+	for i := 0; i < len(cluster); i++ {
+		hosts := govcLoginCmd(vcAddress, vCenterPort) + "govc ls " + cluster[i] + " " + " | grep 10."
+		framework.Logf("Fetch host details: %s ", hosts)
+		hostsResult, err := sshExec(sshClientConfig, masterIp, hosts)
+		if err != nil && hostsResult.Code != 0 {
+			fssh.LogResult(hostsResult)
+			return nil, fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+				hosts, masterIp, err)
+		}
+		if hostsResult.Stdout != "" {
+			hostListTemp := strings.Split(hostsResult.Stdout, "\n")
+			hostList = append(hostList, hostListTemp...)
+		}
+		for i := 0; i < len(hostList); i++ {
+			if hostList[i] != "" {
+				hostsList = append(hostsList, hostList[i])
+			}
+		}
+		hostList = nil
+	}
+	return hostsList, nil
+}
+
+// getVmDetails method is used to fetch vm details
+func getVmDetails(vcAddress string, vCenterPort string, sshClientConfig *ssh.ClientConfig,
+	masterIp string, dataCenter []string) ([]string, error) {
+	var vmsList []string
+	var vmList []string
+	// get vm details
+	ginkgo.By("Executing command for fetching k8s vm details")
+	for i := 0; i < len(dataCenter); i++ {
+		vms := govcLoginCmd(vcAddress, vCenterPort) + "govc ls /" + dataCenter[i] + "/vm" + " " + "| grep 'k8s\\|haproxy'"
+		framework.Logf("Fetch vm details: %s ", vms)
+		vMsResult, err := sshExec(sshClientConfig, masterIp, vms)
+		if err != nil && vMsResult.Code != 0 {
+			fssh.LogResult(vMsResult)
+			return nil, fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+				vms, masterIp, err)
+		}
+		if vMsResult.Stdout != "" {
+			vmListTemp := strings.Split(vMsResult.Stdout, "\n")
+			vmList = append(vmList, vmListTemp...)
+		}
+		for i := 0; i < len(vmList); i++ {
+			if vmList[i] != "" {
+				vmsList = append(vmsList, vmList[i])
+			}
+		}
+		vmList = nil
+	}
+	return vmsList, nil
+}
+
+// getDatastoreDetails method is used to fetch datastore details
+func getDatastoreDetails(vcAddress string, vCenterPort string, sshClientConfig *ssh.ClientConfig,
```

Comment:
you can refer pkg/common/cns-lib/vsphere/virtualcenter.go on how to get objects from VC using govmomi APIs